### DISCUSSION TITLE
Provide a way to access the main route

### DISCUSSION THREAD

**Discussion by dianfishekqi** [livewire/livewire#4031] [1. Help] [answered]
Provide a way to access the main route name to child components.

**Comment by joshhanley** [comment]
@dianfishekqi can you explain in detail what you mean and what you are hoping to achieve? There may already be a way to do this 🙂

Thanks!

**Reply by dianfishekqi** [reply]
@joshhanley Switching teams in jetstream redirects you to home route, is there to get the main route from the team switcher component without passing it through?

Thanks

**Reply by joshhanley** [reply]
@dianfishekqi what do you mean by main route? Do you mean what route you are currently on or the root `/` route?

If you mean the current route, you should be able to get that from the request.

**Reply by dianfishekqi** [reply]
Lets say I'm at `/home` checking the route name inside the team switcher component returns `livewire....` and there is no way to tell the actions is performed from `/home` ...

**Reply by walirazzaq** [reply, marked as answer]
As @joshhanley mentioned you do something like this and create a helper. 
```php
   $request = Request::create(Livewire::originalUrl(), Livewire::originalMethod());
   $route    = app('router')->getRoutes()->match($request);
   $name    = $route->getName();
```
Livewire uses same technique to apply middlewares to the routes
